mongo/integration/client_test.go

```diff
@@ -107,39 +107,48 @@ func TestClient(t *testing.T) {
 		assert.Equal(t, 2, len(split), "expected 2 parts after split, got %v (connstring %v)", split, baseConnString)
 		revisedConnString += split[1]
 
-		cs := fmt.Sprintf(
-			"%s&sslClientCertificateKeyFile=%s&authMechanism=MONGODB-X509&authSource=$external",
-			revisedConnString,
-			path.Join(certificatesDir, "client.pem"),
-		)
-		authClient, err := mongo.Connect(mtest.Background, options.Client().ApplyURI(cs))
-		assert.Nil(mt, err, "authClient Connect error: %v", err)
-		defer func() { _ = authClient.Disconnect(mtest.Background) }()
-
-		rdr, err := authClient.Database("test").RunCommand(mtest.Background, bson.D{
-			{"connectionStatus", 1},
-		}).DecodeBytes()
-		assert.Nil(mt, err, "connectionStatus error: %v", err)
-		users, err := rdr.LookupErr("authInfo", "authenticatedUsers")
-		assert.Nil(mt, err, "authenticatedUsers not found in response")
-		elems, err := users.Array().Elements()
-		assert.Nil(mt, err, "error getting users elements: %v", err)
-
-		for _, userElem := range elems {
-			rdr := userElem.Value().Document()
-			var u struct {
-				User string
-				DB   string
+		certificates := [3]string{"client.pem", "client-pkcs8-encrypted.pem", "client-pkcs8-unencrypted.pem"}
+		for _, certificate := range certificates {
+			keyFilePassword := ""
```

Comment:
nit: For empty declarations, I generally prefer using the zero value: `var keyFilePassword string`

Reply:
Nice! Good to learn common Go patterns :)